assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,247 @@
+
+const baseInitialState = {
+	sharingSettings: undefined,
+	savedSharingSettings: undefined,
+};
+
+const fetchSaveSharingSettingsStore = createFetchStore( {
+	baseName: 'saveSharingSettings',
+	controlCallback: ( { savedSharingSettings } ) => {
+		return API.set(
+			'core',
+			'modules',
+			'sharing-settings',
+			savedSharingSettings,
+			{
+				useCache: false,
+			}
+		);
+	},
+	reducerCallback: ( state, { settings } ) => {
+		return {
+			...state,
+			savedSharingSettings: settings,
+			sharingSettings: settings,
+		};
+	},
+	argsToParams: ( savedSharingSettings ) => ( { savedSharingSettings } ),
+	validateParams: ( { savedSharingSettings } = {} ) => {
+		invariant( savedSharingSettings, 'savedSharingSettings is required.' );
+	},
+} );
+
+const baseActions = {
+	/**
+	 * Sets the sharing settings management role of given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} moduleSlug Module slug.
+	 * @param {string} management New management role for module, one of: all_admins, owner.
+	 * @return {Object} Action for SET_SHARING_MANAGEMENT.
+	 */
+	setSharingManagement( moduleSlug, management ) {
+		invariant( moduleSlug, 'moduleSlug is required.' );
+		const managementRoles = [ 'all_admins', 'owner' ];
+		invariant(
+			managementRoles.includes( management ),
+			`management must be one of: ${ managementRoles.join( ', ' ) }.`
+		);
+		return {
+			payload: {
+				moduleSlug,
+				management,
+			},
+			type: SET_SHARING_MANAGEMENT,
+		};
+	},
+
+	/**
+	 * Sets the sharing settings management role of given module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string}   moduleSlug Module slug.
+	 * @param {string[]} roles      List of roles the module is shared with.
+	 * @return {Object} Action for SET_SHARED_ROLES.
+	 */
+	setSharedRoles( moduleSlug, roles ) {
+		invariant( moduleSlug, 'moduleSlug is required.' );
+		invariant(
+			Array.isArray( roles ) &&
+				roles.every( ( role ) => typeof role === 'string' ),
+			'roles must be an array of strings.'
+		);
+		return {
+			payload: {
+				moduleSlug,
+				roles,
+			},
+			type: SET_SHARED_ROLES,
+		};
+	},
+
+	/**
+	 * Saves sharingSettings for dashboard sharing.
+	 *
+	 * Save sharingSettings for dashboard sharing.
+	 * Update ownerID from the response for the modules in the sharingSettings state.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {Object} Object with `{response, error}`.
+	 */
+	*saveSharingSettings() {
+		const registry = yield Data.commonActions.getRegistry();
+		// TODO: Refactor to use `getSharingSettings` selector
+		// to obtain the `sharingSettings` from the state in 4795.
+		const { sharingSettings } = registry.stores[
+			CORE_MODULES
+		].store.getState();
+
+		const {
+			response,
+			error,
+		} = yield fetchSaveSharingSettingsStore.actions.fetchSaveSharingSettings(
+			sharingSettings
+		);
+
+		// Update module owner IDs in the sharing settings modules.
+		if ( Object.keys( response?.newOwnerIDs ).length ) {
```

Comment:
We don't need the conditional chain for accessing `newOwnerIDs` as it will always be in the response, but also if it were missing this code would still raise an error because you can't get keys from `undefined`
![image](https://user-images.githubusercontent.com/1621608/170103233-2339eff9-f1d9-43b1-bca9-1f2a7ac72efa.png)

Instead, we need to first check that there is no `error` after which `response.newOwnerIDs` can be safely accessed.